### PR TITLE
Feature: Unit Tests for Methods used in CertificateAuthorityService

### PR DESCRIPTION
**Pull request by lydiazcheng**
This PR is to add unit tests for methods which are used in CertificateAuthorityService but belongs to other scripts, including `Acme`, `Hubject` and `CertificateUtil`. The tests for CertificateAuthorityService itself will be in the next PR.

It also fix/improve few places in the source code and add test coverage command in the project as below.
Screenshot 2024-08-19 at 10 17 41 AM

There is an 404 error when using `signCertificateByExternalCA` in `Acme` so unit test is not added for this method. We use self-signed certificate currently. I created a ticket for this issue and we can plan and fix it later.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
     "fresh-and-install-all": "npm run fresh && npm run install-all",
     "sync-db": "ts-node ./db.sync.ts",
     "start-everest": "npm run start-everest --prefix ./Server",
-    "test": "jest --config jest.config.ts"
+    "test": "jest --config jest.config.ts",
+    "coverage": "jest --config jest.config.ts --coverage"
```
Comment (ChrisWeissmann) [inline package.json:36]: nice! love it!

File: Server/src/config/envs/local.ts [hunk 2]
```diff
@@ -81,6 +81,7 @@ export function createLocalConfig() {
         host: '0.0.0.0',
         port: 8055,
         generateFlows: false,
+        token: '-ssaT85n4S-wVD21LKOCDwvXN5PtnJc0',
```
Comment (ChrisWeissmann) [inline Server/src/config/envs/local.ts:84]: Is this token needed?

File: 02_Util/test/certificate/client/Acme.test.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            name: 'acme',
+            acme: {
+              env: 'staging',
+              accountKeyFilePath: faker.lorem.word(),
+            },
+          },
+        },
+      },
+    } as any;
+    logger = new Logger<ILogObj>({ name: expect.getState().currentTestName });
```
Comment (ChrisWeissmann) [inline 02_Util/test/certificate/client/Acme.test.ts:56]: I think you could leave out the logger even. the Acme class instantiates it as it's optional.

File: 02_Util/test/certificate/client/Hubject.test.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            hubject: {
+              baseUrl: mockBaseURL,
+              tokenUrl: mockTokenURL,
+              isoVersion: mockISOVersion,
+            },
+          },
+        },
+      },
+    } as any;
+    logger = new Logger<ILogObj>({ name: expect.getState().currentTestName });
```
Comment (ChrisWeissmann) [inline 02_Util/test/certificate/client/Hubject.test.ts:40]: In theory you don't need the logger I believe

**Review by ChrisWeissmann — APPROVED** [review]
Just some minor nitpicks

**Comment by lydiazcheng** [comment]
> Just some minor nitpicks

Hi Christian, thanks a lot for helping review this PR!

File: Server/src/config/envs/local.ts (hunk 2 above)
Comment (lydiazcheng) [reply Server/src/config/envs/local.ts:84]: Yes. This static token is used when core creats directus client. It is here: https://github.com/citrineos/citrineos-core/blob/823849c04e2a58212e9facff946496cdde218f5d/02_Util/src/util/directus.ts#L42 Then directus client works when, e.g., upload files. I updated it in docker config before but forgot local. 😞  https://github.com/citrineos/citrineos-core/blob/823849c04e2a58212e9facff946496cdde218f5d/Server/docker-compose.yml#L84

**Pull request merged**